core/src/test/java/org/elasticsearch/cluster/routing/allocation/FailedShardsRoutingTests.java

```diff
@@ -218,9 +220,6 @@ public void testFailPrimaryStartedCheckReplicaElected() {
         assertThat(routingTable.index("test").shard(0).primaryShard().currentNodeId(), anyOf(equalTo("node1"), equalTo("node2")));
         assertThat(routingTable.index("test").shard(0).replicaShards().size(), equalTo(1));
         assertThat(routingTable.index("test").shard(0).replicaShards().get(0).state(), equalTo(UNASSIGNED));
-
-        logger.info("fail the shard again, check that nothing happens");
```

Comment:
this made go and check that this is tested in ShardStateActionTests , which it is, but we miss test there, like double sending the same shard before it's processed and failing a shard of an index that doesn't exist